sw/device/lib/dif/dif_entropy.h

```diff
@@ -0,0 +1,678 @@
+} dif_entropy_test_stats_t;
+
+/**
+ * The result of an entropy source operation.
+ */
+typedef enum dif_entropy_result {
+  /**
+   * Indicates that the operation succeeded.
+   */
+  kDifEntropyOk = 0,
+  /**
+   * Indicates some unspecified failure.
+   */
+  kDifEntropyError = 1,
+  /**
+   * Indicates that some parameter passed into a function failed a
+   * precondition.
+   *
+   * When this value is returned, no hardware operations occured.
+   */
+  kDifEntropyBadArg = 2,
+  /**
+   * Indicates that this operation has been locked out, and can never
+   * succeed until hardware reset.
+   */
+  // Remove this variant if you don't need it.
+  kDifEntropyLocked = 3,
+} dif_entropy_result_t;
+
+/**
+ * An entropy source interrupt request type.
+ */
+typedef enum dif_entropy_irq {
+  /**
+   * Indicates that bits of entropy are available to consume.
+   */
+  kDifEntropyIrqAvailable,
+
+  /**
+   * Indicates that the health test has failed and the alert count has been
+   * met.
+   */
+  kDifEntropyIrqUnhealthy,
+
+  /**
+   * Indicates that an internal error occured in the FIFO.
+   */
+  kDifEntropyIrqFifoError,
+} dif_entropy_irq_t;
+
+/**
+ * A snapshot of the enablement state of the interrupts for an entropy source.
+ *
+ * This is an opaque type, to be used with the `dif_entropy_irq_disable_all()`
+ * and `dif_entropy_irq_restore_all()` functions.
+ */
+typedef uint32_t dif_entropy_irq_snapshot_t;
+
+/**
+ * An entropy source alert type.
+ */
+typedef enum dif_entropy_alert {
+  /**
+   * Indicates that the health test criteria were not met.
+   */
+  kDifEntropyAlert,
+} dif_entropy_alert_t;
+
+/**
+ * Creates a new handle for entropy source.
+ *
+ * This function does not actuate the hardware.
+ *
+ * @param params Hardware instantiation parameters.
+ * @param[out] entropy Out param for the initialized handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_init(dif_entropy_params_t params,
+                                      dif_entropy_t *entropy);
+
+/**
+ * Configures entropy source with runtime information.
+ *
+ * This function should need to be called once for the lifetime of `handle`.
+ *
+ * @param entropy An entropy source handle.
+ * @param config Runtime configuration parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_configure(const dif_entropy_t *entropy,
+                                           dif_entropy_config_t config);
+
+/**
+ * Queries the entropy source IP for its revision information.
+ *
+ * @param entropy An entropy source handle.
+ * @param[out] revision Out-param for revision data.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_entropy_result_t dif_entropy_get_revision(const dif_entropy_t *entropy,
+                                              dif_entropy_revision_t *revision);
+
+/**
+ * Queries the entropy source for health statistics.
+ *
```

Comment:
nit: I guess this will also clear relevant registers.

Reply:
sorry, I think I misunderstood your comment. I added the note you recommended re. clearing relevant registers. thanks!